### PR TITLE
restructured api with new Connection interface

### PR DESCRIPTION
**Pull request by huiminliu09**
Chang some of the with Connection( ) to  Connection.sql_execute( ) in 
- tweet_router:
     - send_recent_tweet_data()
- search_router:
     - search_administrative_boundaries()
- data_router:
     - region_temp()
     - region_moisture()

### CODE REVIEW

File: backend/web/router/search_router.py [hunk 1]
```diff
@@ -31,13 +31,10 @@ def search_administrative_boundaries():
...
         '''
 
-        with Connection() as conn:
-            cur = conn.cursor()
-            cur.execute(query)
-            resp = make_response(jsonify(
-                [json.loads(geom) for geom, in cur.fetchall()]
+        return make_response(jsonify(
+            [json.loads(geom) for geom, in Connection.sql_execute(query)]
             ))
```
Comment (Yicong-Huang) [inline backend/web/router/search_router.py]: make this in one line

File: backend/web/router/data_router.py [hunk 2]
```diff
@@ -184,14 +184,10 @@ def region_temp():
     and noaa.tid = rft.tid
     GROUP BY date(rft.reftime)
     '''
+    return make_response(jsonify(
+        fill_series(date_series,
+                    Connection.sql_execute(query.format(region_id=region_id, timestamp=timestamp_str, days=days)))))
```
Comment (Yicong-Huang) [inline backend/web/router/data_router.py]: change it to f-string instead of str.format

File: backend/web/__init__.py [hunk 3]
```diff
@@ -1,44 +1,44 @@
...
+    # register routers
+    app.register_blueprint(router.search_router.bp)
+    app.register_blueprint(router.data_router.bp)
+    app.register_blueprint(router.tweet_router.bp)
+    app.register_blueprint(router.root_router.bp)
+    app.register_blueprint(router.dropDownMenu_router.bp)
+
+    return app
+
+
```
Comment (Yicong-Huang) [inline backend/web/__init__.py]: what is the change in this file?

**Review by Yicong-Huang — CHANGES_REQUESTED** [review]
I left comments for some minor changes, please take care of them.

File: backend/web/router/data_router.py (hunk 2 above)
Comment (huiminliu09) [reply backend/web/router/data_router.py]: I have changed them according to your comments.

**Comment by Yicong-Huang** [comment]
is this PR ready for review?

**Comment by huiminliu09** [comment]
> is this PR ready for review?
Yes.

File: backend/web/router/data_router.py [hunk 4]
```diff
@@ -231,13 +226,8 @@ def region_moisture():
...
 
-    with Connection() as conn:
-        cur = conn.cursor()
-        cur.execute(query.format(region_id=region_id, timestamp=timestamp_str, days=days))
-        resp = make_response(jsonify(
-            fill_series(date_series, cur.fetchall())
-        ))
-    return resp
+    return make_response(jsonify(
+        fill_series(date_series, Connection.sql_execute(query))))
```
Comment (Yicong-Huang) [inline backend/web/router/data_router.py:229]: make it one line

File: backend/web/router/root_router.py [hunk 5]
```diff
@@ -28,10 +28,10 @@ def send_wildfire():
...
         jsonify([{"long": lon, "lat": lat, "nlp": True, "text": text} for lon, lat, nlp_text, text in
-                 Connection().sql_execute(
+                 Connection.sql_execute(
                      f"select l.top_left_long, l.top_left_lat, r.text, r.text from locations l, images i, records r "
                      f"where l.id = i.id and r.id = l.id and i.wildfire_prob>0.9 and "
                      f"l.top_left_long>{west} and l.top_left_lat<{north} "
                      f"and l.bottom_right_long<{east} and l.bottom_right_lat>{south} "
                      f"and extract(epoch from r.create_at) >{start} and extract(epoch from r.create_at) <{end}")]))
-    return resp
+    return resp
```
Comment (Yicong-Huang) [inline backend/web/router/root_router.py]: make it inline (remove the resp declaration and just return the make_response)

File: backend/web/router/tweet_router.py [hunk 6]
```diff
@@ -84,7 +83,7 @@ def send_fire_tweet_data():
     """
     resp = make_response(
         jsonify([{"create_at": time.isoformat(), "long": lon, "lat": lat, "id": str(id)} for time, lon, lat, _, _, id in
-                 Connection().sql_execute(
+                 Connection.sql_execute(
                      "select r.create_at, l.top_left_long, l.top_left_lat, l.bottom_right_long, l.bottom_right_lat, r.id "
                      "from records r,locations l where r.id=l.id AND create_at>now()-interval '30 day'")]))
     return resp
```
Comment (Yicong-Huang) [inline backend/web/router/tweet_router.py]: make it inline

File: backend/web/router/tweet_router.py [hunk 7]
```diff
@@ -189,7 +182,7 @@ def tweet_by_date():
...
```
Comment (Yicong-Huang) [inline backend/web/router/tweet_router.py]: make it inline

**Review by Yicong-Huang — CHANGES_REQUESTED** [review]
some statements can be in line, please fix them. will merge this after this fix

**Review by Yicong-Huang — APPROVED** [review]
thanks for the PR

**Pull request merged**